doc/api/single-executable-applications.md

```diff
@@ -0,0 +1,104 @@
+# Single executable applications
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental: This feature is currently being designed and will
+> change.
+
+<!-- source_link=lib/internal/main/single_executable_application.js -->
+
+This feature allows the distribution of a Node.js application conveniently to a
+system that does not have Node.js installed.
+
+Node.js supports the creation of [single executable applications][] by allowing
+the injection of a JavaScript file into the `node` binary. During start up, the
+program checks if anything has been injected. If the script is found, it
+executes its contents, otherwise it operates like plain Node.js.
+
+A bundled JavaScript file can be turned into a single executable application
+with any tool which can inject resources into the `node` binary.
+
+One such tool is [postject][]:
+
+```console
+$ cat hello.js
+console.log(`Hello, ${process.argv[2]}!`);
+$ cp $(command -v node) hello
+# On systems other than macOS:
+$ npx postject hello NODE_JS_CODE hello.js \
+    --sentinel-fuse NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2
+# On macOS:
+$ npx postject hello NODE_JS_CODE hello.js \
+    --sentinel-fuse NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2 \
+    --macho-segment-name NODE_JS
+$ ./hello world
+Hello, world!
+```
+
+This currently only supports running a single embedded [CommonJS][] file.
```

Comment:
> the single executable application feature whether or not you use `postject`

This one ^. Should we adjust it accordingly then? I keep the "currently" bit because I do plan to add support for ESM files, etc.

Reply:
Perhaps this?

```suggestion
Single executable applications only support running a single embedded [CommonJS][] file.
```

If you want to keep `currently`, you can, but as a reader of the docs, I would prefer to know what is planned. (Is it multiple CommonJS files? ESM? Both? Something else?)

```suggestion
Single executable applications only support running a single embedded
[CommonJS][] file. We plan to introduce support for multiple CommonJS
files and [ESM][] future versions of Node.js.
```

At the same time, the docs apply to the API for specific versions of Node.js, so there's not really a need to talk about future versions in the API docs in my opinion. (But also, our API docs also function a bit as guides for end users and not merely API references, so talking about that kind of context can be unavoidable. So I see why you'd want to include it for sure and don't oppose it. Mild preference to not have that kind of stuff, but like I said, it can be unavoidable.)

Reply:
Hmm, fair point. The `Upcoming features` section of `doc/contributing/maintaining-single-executable-application-support.md` already talks about future plans, so I also think it's okay to remove the "currently" bit then. :+1: